cmd/signimage/signimage.go

```diff
@@ -0,0 +1,309 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"flag"
+	"fmt"
+	"github.com/docker/cli/cli/config"
+	dockertypes "github.com/docker/cli/cli/config/types"
+	"github.com/go-logr/logr"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/registry"
+	"io"
+	"k8s.io/klog/v2/klogr"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func checkArg(arg *string, varname string, fallback string) {
+	if *arg == "" {
+		if fallback != "" {
+			*arg = fallback
+		} else {
+			fmt.Printf("%s not found:\n", varname)
+			flag.PrintDefaults()
+			os.Exit(0)
+		}
+	}
+	return
+}
+
+func canonicalisePath(path string) string {
+	canonical := strings.Replace("/"+path, "/./", "/", -1)
+	return strings.Replace(canonical, "//", "/", -1)
+}
+
+func extractFile(tmpdir string, header *tar.Header, tarreader io.Reader) (string, error) {
```

Comment:
why not make this function part of registry package?

Reply:
We'd end up with signimage calling into registry which calls back into signimage, which in turn calls back into registry, which feels complicated.

But then additionally extractFile is pretty much tailored for the signimage use case (it expects io.Reader to be lined up ready to read the first file, then extracts to a specific place in /tmp that maintains the path information so it can be added to the new layer, and creates things with specific permissions), some of this could be factored out into registry but its not generic enough to all go in there, 

So the code is simple if its in signimage but moving those few lines to registry makes it a lot more complex.

Reply:
besides persmissions, which can be passed as a parameter, i don't see anything sign-related code in this function. io.Reader can be set to whatever offset user want prior to calling function, and extracting file to user-defined directory also seems ok for general use case

Reply:
Ok, I've moved that functionality to the registry library. I think it makes the signimage code worse because it has to pass the registry object into itself, so it can be passed back to signimage, so it can call functions in registry, but I guess if anyone else ever uses the registry library for extracting files it will help them.